components/client/src/test/unit/java/com/hotels/styx/client/loadbalancing/strategies/PowerOfTwoStrategyTest.java

```diff
@@ -0,0 +1,121 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.client.loadbalancing.strategies;
+
+import com.google.common.collect.Iterables;
+import com.hotels.styx.api.HttpClient;
+import com.hotels.styx.api.client.ActiveOrigins;
+import com.hotels.styx.api.client.ConnectionPool;
+import com.hotels.styx.api.client.Origin;
+import com.hotels.styx.api.client.RemoteHost;
+import com.hotels.styx.api.client.loadbalancing.spi.LoadBalancingStrategy;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+
+import static com.google.common.collect.Iterables.getFirst;
+import static com.hotels.styx.api.client.Origin.newOriginBuilder;
+import static com.hotels.styx.api.client.RemoteHost.remoteHost;
+import static com.hotels.styx.api.support.HostAndPorts.localHostAndFreePort;
+import static java.util.Arrays.asList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+public class PowerOfTwoStrategyTest {
+    final Origin ORIGIN_ONE = newOriginBuilder(localHostAndFreePort()).id("one").build();
+    final Origin ORIGIN_TWO = newOriginBuilder(localHostAndFreePort()).id("two").build();
+    final Origin ORIGIN_THREE = newOriginBuilder(localHostAndFreePort()).id("three").build();
+    final Origin ORIGIN_FOUR = newOriginBuilder(localHostAndFreePort()).id("four").build();
+
+    private final RemoteHost HOST_ONE = remoteHost(ORIGIN_ONE, mockPool(5), mock(HttpClient.class));
+    private final RemoteHost HOST_TWO = remoteHost(ORIGIN_TWO, mockPool(6), mock(HttpClient.class));
+    private final RemoteHost HOST_THREE = remoteHost(ORIGIN_THREE, mockPool(3), mock(HttpClient.class));
+    private final RemoteHost HOST_FOUR = remoteHost(ORIGIN_FOUR, mockPool(2), mock(HttpClient.class));
+
+    List<RemoteHost> allOrigins = asList(HOST_ONE, HOST_TWO, HOST_THREE,  HOST_FOUR);
+
+    private long RNG_SEED = 5;
+
+
+    @Test
+    public void choosesBetterOfTwoRandomChoices() {
+        ActiveOrigins activeOrigins = mock(ActiveOrigins.class);
+        when(activeOrigins.snapshot()).thenReturn(allOrigins);
+
+        Random rng = new Random(RNG_SEED);
+        int first = rng.nextInt(4);
+        assert(first == 2);
+
+        int second = rng.nextInt(4);
+        assert(second == 0);
+
+        PowerOfTwoStrategy loadBalancer = new PowerOfTwoStrategy(activeOrigins, new Random(RNG_SEED));
+        Iterable<RemoteHost> chosenOne = loadBalancer.vote(mock(LoadBalancingStrategy.Context.class));
+
+        assertThat(getFirst(chosenOne, null), is(betterOf(allOrigins.get(first), allOrigins.get(second))));
```

Comment:
It would be clearer to have the expected result stated explicitly, rather than generated by a function.
We could add a brief (one line) comment to explain why it is the expected result if necessary.